Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -134,36 +145,39 @@ IRelocator relocator
 
       databaseEntry.CancelPendingUpdates();
       var cancellationSource = new CancellationTokenSource();
-      var previousDocumentTask = databaseEntry.LatestDocument;
-      var resolvedDocumentTask = ApplyChangesAsync(previousDocumentTask, documentChange, cancellationSource.Token);
+      var updatedText = textChangeProcessor.ApplyChange(databaseEntry.TextBuffer, documentChange, CancellationToken.None);
```

Comment:
Excellent ! This removes the computation of the updated text to be in a task, which was probably a cause of concurrent issues such as https://github.com/dafny-lang/dafny/issues/2246

This also means that the TextBufferItem is now shared between the `DocumentEntry` and the `DafnyDocument`, with the second possibly outdated.
Can you confirm you ensures that `UpdateDocument` itself will never fail, so that we are immediately left with the newest version of the document every time?